distributed/tests/test_worker.py

```diff
@@ -2837,6 +2845,32 @@ async def test_acquire_replicas_many(c, s, *workers):
         await asyncio.sleep(0.001)
 
 
+@pytest.mark.slow
+@gen_cluster(client=True, Worker=Nanny)
+async def test_acquire_replicas_already_in_flight(c, s, *nannies):
+    """Trying to acquire a replica that is already in flight is a no-op"""
+
+    class SlowToFly:
+        def __getstate__(self):
+            sleep(1.5)
+            return {}
+
+    a, b = s.workers
+    x = c.submit(SlowToFly, workers=[a], key="x")
+    await wait(x)
+    y = c.submit(lambda x: x, x, workers=[b], key="y")
+    await asyncio.sleep(0.5)
+    start = time()
+    _acquire_replicas(s, b, x)
+    await asyncio.sleep(0.5)
+
+    story = await c.run(lambda dask_worker: dask_worker.story("x"), workers=[b])
+    events = [ev for ev in story[b] if ev[-1] >= start]
+    assert len(events) == 2
+    assert events[0][:3] == ("x", "ensure-task-exists", "flight")
+    assert events[1][:4] == ("x", "flight", "fetch", "flight")
```

Comment:
Not necessary for this PR but rather a comment for the future.

I often find myself getting the story of a task just to filter it later on, e.g. removing stimulus IDs for tests. We should consider adding that as a functionality to `Worker.story`. I'm wondering what other use cases are helpful there